File: markdown/src/main/java/ru/noties/markwon/extension/emoji/EmojiExtension.java
package ru.noties.markwon.extension.emoji;

import org.commonmark.Extension;
import org.commonmark.parser.Parser;
import org.commonmark.renderer.html.HtmlRenderer;

import ru.noties.markwon.extension.emoji.internal.EmojiDelimiterProcessor;
import ru.noties.markwon.extension.emoji.internal.EmojiNodeRenderer;

/**
 * Created by kosh on 20/08/2017.
 */

public class EmojiExtension implements Parser.ParserExtension, HtmlRenderer.HtmlRendererExtension {
    private EmojiExtension() {}

    public static Extension create() {
        return new EmojiExtension();
    }

    @Override public void extend(Parser.Builder parserBuilder) {
        parserBuilder.customDelimiterProcessor(new EmojiDelimiterProcessor());
    }

    @Override public void extend(HtmlRenderer.Builder rendererBuilder) {
        rendererBuilder.nodeRendererFactory(EmojiNodeRenderer::new);
    }
}

File: app/src/main/java/com/fastaccess/github/platform/markwon/emoji/EmojiExtension.kt
package com.fastaccess.github.platform.markwon.emoji

import com.fastaccess.github.platform.markwon.emoji.internal.EmojiDelimiterProcessor
import com.fastaccess.github.platform.markwon.emoji.internal.EmojiNodeRenderer
import org.commonmark.Extension
import org.commonmark.parser.Parser
import org.commonmark.renderer.html.HtmlRenderer

/**
 * Created by kosh on 20/08/2017.
 */

class EmojiExtension private constructor() : Parser.ParserExtension, HtmlRenderer.HtmlRendererExtension {

    override fun extend(parserBuilder: Parser.Builder) {
        parserBuilder.customDelimiterProcessor(EmojiDelimiterProcessor())
    }

    override fun extend(rendererBuilder: HtmlRenderer.Builder) {
        rendererBuilder.nodeRendererFactory { EmojiNodeRenderer(it) }
    }

    companion object {

        fun create(): Extension {
            return EmojiExtension()
        }
    }
}
